### ISSUE TITLE
Patch Secrets with new data, instead of appending.

### ISSUE THREAD

**Issue by cpressland** [kr8s-org/kr8s#201]
### Which project are you requesting an enhancement for?

kr8s

### What do you need?

Currently `kr8s` exposes a `.patch()` mechanism for Secret objects which allows us to add new keys:

```python
>>> from kr8s.objects import Secret
>>> a = Secret("my-secret")
>>> a.refresh()
>>> a.raw["data"]
{"aaa": "ZXhhbXBsZQ=="}
>>> a.patch({"data": {"bbb": "bW9yZV9leGFtcGxl"}})
>>> a.raw["data"]
{"aaa": "ZXhhbXBsZQ==", "bbb": "bW9yZV9leGFtcGxl"}
```

I'd like the ability to set a `replace=True` arg to the `.patch()` call to have the secret data set to the contents of the patch call. Example:

```python
>>> from kr8s.objects import Secret
>>> a = Secret("my-secret")
>>> a.refresh()
>>> a.raw["data"]
{"aaa": "ZXhhbXBsZQ=="}
>>> a.patch({"data": {"bbb": "bW9yZV9leGFtcGxl"}}, replace=True)
>>> a.raw["data"]
{"bbb": "bW9yZV9leGFtcGxl"}
```

Alternatively, supporting JSON 6902 style patching would also be a valid solution here, similar to how kubectl handles this:

```shell
$ kubectl patch secret my-secret \
    --type='json' \
    -p='[{"op": "replace", "path": "/data", "value":{"bbb": "bW9yZV9leGFtcGxl"}}]'
```

This could be implemented as a new function:

```python
>>> a.patch6902([{"op": "replace", "path": "/data", "value":{"bbb": "bW9yZV9leGFtcGxl"}}])
>>> a.raw["data"]
{"bbb": "bW9yZV9leGFtcGxl"}
```

**Comment by jacobtomlinson** [comment]
Thanks for raising this @cpressland. Given our goal of being similar to `kubectl` I think supporting the JSON 6902 style would make a lot of sense.

Perhaps with an API like this would feel natural to `kubectl` users?

```python
a.patch([{"op": "replace", "path": "/data", "value":{"bbb": "bW9yZV9leGFtcGxl"}}], type="json")
```

In #75 I've also been exploring implementing `apply` which along with a setter on `Secret.data` could provide an API like this.

```python
a.data = {"bbb": "bW9yZV9leGFtcGxl"}
a.apply()
```

**Comment by cpressland** [comment]
Both options look completely valid, I'd go further and suggest both approaches should be implemented, option 1 because it's what a kubectl user would expect to "just work", and option 2 because it looks more pythonic and provides a shorthand solution for small cases like this.

**Comment by jacobtomlinson** [comment]
Yup totally agree. The apply option is already in hand as part of that larger PR. For the JSON 6902 implementation, I expect this is just the case of adding an extra header when we make the API call. Do you have any interest in contributing this @cpressland? Otherwise I'll take a look when I can.

**Comment by jacobtomlinson** [comment]
Thanks for the offer, it actually turned out to be a small change so I threw #202 together.

**Comment by jacobtomlinson** [comment]
JSON 6902 style patching is now available in v0.10.0
